test/e2e/auth/projected_clustertrustbundle.go

```diff
@@ -268,11 +268,11 @@ var _ = SIGDescribe(feature.ClusterTrustBundle, feature.ClusterTrustBundleProjec
 		var cleanups []func(ctx context.Context)
 		var projections []v1.VolumeProjection
 
-		defer func() {
```

Comment:
this is added to that hook at the beginning, see line 65

```
	ginkgo.JustBeforeEach(func(ctx context.Context) {
		cleanup := mustInitCTBs(ctx, f, initCTBs)
		ginkgo.DeferCleanup(cleanup)
	})
```

Reply:
@aojea The `initCTBs` and `cleanups` are declared in the `It` container, not in its parent container `Describe`. 

- `child` https://github.com/carlory/kubernetes/blob/master/test/e2e/auth/projected_clustertrustbundle.go#L268
- `parent` https://github.com/carlory/kubernetes/blob/master/test/e2e/auth/projected_clustertrustbundle.go#L64

so `JustBeforeEach` won't call `cleanups`.

BTW, in my opinion, the `JustBeforeEach` is abused here. It's used to separate creation and configuration which is set with the `BeforeEach` in each child `It` container. `JustBeforeEach` should be used with care as it can add complexity to a test suite.

> While powerful and flexible overuse of JustBeforeEach (and nest JustBeforeEaches in particular!) can lead to confusing suites to be sure to use JustBeforeEach judiciously!
> From the doc https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach 

`BeforeEach` may be enough for this e2e suite.